src/Application/Features/QualityAssurance/Commands/SubmitPqaResponse.cs

```diff
@@ -147,16 +147,45 @@ public E_OwnerShouldNotBeApprover(IUnitOfWork unitOfWork)
             _unitOfWork = unitOfWork;
 
             RuleFor(c => c)
-                .MustAsync(OwnerMustNotBeApprover)
+                .Must(OwnerMustNotBeApprover)
                 .WithMessage("This assessment is created by you hence must not be processed at PQA stage by you");
         }
 
-        private async Task<bool> OwnerMustNotBeApprover(Command c, CancellationToken cancellationToken)
+        private bool OwnerMustNotBeApprover(Command c)
         {
-            var entry = await _unitOfWork.DbContext.EnrolmentPqaQueue.Include(c => c.Participant)
-                .FirstOrDefaultAsync(a => a.Id == c.QueueEntryId, cancellationToken: cancellationToken);
+            var entry = _unitOfWork.DbContext.EnrolmentPqaQueue.Include(c => c.Participant)
+                .FirstOrDefault(a => a.Id == c.QueueEntryId);
 
             return entry != null && entry.Participant!.OwnerId!.Equals(c.CurrentUser!.UserId) == false;
         }
     }
+
+    public class G_EnrolmentOccurredWithin3Months : AbstractValidator<Command>
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        public G_EnrolmentOccurredWithin3Months(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+
+            RuleFor(g => g)
+                .Must(EnrolmentOccurredWithin3Months)
+                .WithMessage("The enrolment consent date is over 3 months ago")
+                .When(g => g.Response == PqaResponse.Accept);
+        }
+
+        private bool EnrolmentOccurredWithin3Months(Command c)
+        {
+            if (c.Response == PqaResponse.Accept)
+            {
+                var entry = _unitOfWork.DbContext.EnrolmentPqaQueue
+                    .Include(c => c.Participant)
+                    .Include(d => d.Participant!.Consents)
+                    .FirstOrDefault(a => a.Id == c.QueueEntryId);
+
+                return entry != null && entry.Participant!.Consents.Max(a =>a.Lifetime.StartDate) >= DateTime.UtcNow.AddDays(-90);                
+            }
+
+            return false;
```

Comment:
No need to check status, as your `When` clause already does.

```suggestion
           var entry = _unitOfWork.DbContext.EnrolmentPqaQueue
                      .Include(c => c.Participant)
                      .Include(d => d.Participant!.Consents)
                      .FirstOrDefault(a => a.Id == c.QueueEntryId);
                      
                      return entry != null && entry.Participant!.Consents.Max(a =>a.Lifetime.StartDate) >= DateTime.UtcNow.AddDays(-90);                
```